### PR TITLE
E2e test

### PR DESCRIPTION
**Pull request by AnthonyLaw**
### Implementation 
- using cypress for e2e testing
- integrate with Travis CI

### Test-case
- added Header component
- added Footer component
- added Home page
- added Search component
- added Menu router

### CODE REVIEW

File: e2e/components/Footer.spec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+describe('NEM 2 Explorer Footer', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: **NEM2 Explorer Footer should**

File: e2e/components/Footer.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    beforeEach(() => {
+        cy.visit('/')
+    })
+
+    it('click on NEM News', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct nemflash.io given click on NEM News`

File: e2e/components/Footer.spec.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer').contains('NEM News')
+        .click()
+        .should('have.attr', 'href')
+        .and('include', 'nemflash.io')
+    })
+
+    it('click on Forum', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct forum.nem.io given click on Forum`

File: e2e/components/Footer.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Forum')
+        .click()
+        .should('have.attr', 'href')
+        .and('include', 'forum.nem.io')
+    })
+
+    it('click on Github', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct nemtech github given click on Github`

File: e2e/components/Footer.spec.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Github')
+        .click()
+        .should('have.attr', 'href')
+        .and('include', 'github.com/nemtech')
+    })
+
+    it('click on Telegram', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct nemred telegram given click on Telegram`

File: e2e/components/Footer.spec.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Telegram')
+        .click()
+        .should('have.attr', 'href')
+        .and('include', 't.me/nemred')
+    })
+
+    it('click on Reddit', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct /r/nem reddit given click on Reddit`

File: e2e/components/Footer.spec.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Reddit')
+        .click()
+        .should('have.attr', 'href')
+        .and('include', 'reddit.com/r/nem')
+    })
+
+    it('click on Terms', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct terms page given click on Terms`

File: e2e/components/Footer.spec.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Terms')
+        .click()
+
+        cy.url()
+        .should('contain', '/terms')
+    })
+
+    it('click on Privacy', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct privacy page given click on Privacy`

File: e2e/components/Footer.spec.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        cy.get('.footer')
+        .contains('Privacy')
+        .click()
+
+        cy.url()
+        .should('contain', '/privacy')
+    })
+
+    it('click on Node Selector', () => {
```
Comment (evias) [inline e2e/components/Footer.spec.js]: `link to correct ... given click on Node Selector`

OR 

`display node selector given click on Node Selector` 

depending on which test this is supposed to hold.

File: e2e/components/Header.spec.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
+describe('NEM 2 Explorer Header', () => {
```
Comment (evias) [inline e2e/components/Header.spec.js]: **NEM2 Explorer Header should**

File: e2e/components/Header.spec.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    beforeEach(() => {
+        cy.visit('/')
+    })
+
+    it('check on Title', () => {
```
Comment (evias) [inline e2e/components/Header.spec.js]: please apply same text style as above messages: `contain correct logo and title given click on Title`

File: src/components/layout/Header.vue [hunk 12]
```diff
@@ -9,17 +9,17 @@
...
                     </div>
                     <div class="header-sub-title">
                         Search transactions, addresses, namespace & mosaic on the nem network.
                     </div>
-                    <SearchBox />
+                    <SearchBox class='search-box' />
                 </b-col>
                 <b-col md="3" class="header-left">
-                    <LanguageSelector class="d-none d-md-block"/>
+                    <LanguageSelector class="d-none d-md-block language-selector"/>
```
Comment (evias) [inline src/components/layout/Header.vue]: seems like an unnecessary overhead to have "language-selector" wording doubled.

Comment (AnthonyLaw) [reply src/components/layout/Header.vue]: language-selector is added as the className.
It use for **Cypress** to retrieve the the dom element.

**Pull request merged**